### PR TITLE
Add support to publish Gateway URL to Moesif

### PR DESCRIPTION
**Pull request by Lasith650**
Related with: https://github.com/wso2-enterprise/choreo/issues/21887

### CODE REVIEW

File: adapter/internal/oasparser/envoyconf/access_loggers.go [hunk 1]
```diff
@@ -108,6 +108,7 @@ func getGRPCAccessLogConfigs(conf *config.Config) *config_access_logv3.AccessLog
 				Timeout: ptypes.DurationProto(conf.Analytics.Adapter.GRPCRequestTimeout),
 			},
 		},
+		AdditionalRequestHeadersToLog: []string{"x-original-gw-url"},
```
Comment (VirajSalaka) [inline adapter/internal/oasparser/envoyconf/access_loggers.go:111]: Shall we have a constant here too?

Comment (VirajSalaka) [reply adapter/internal/oasparser/envoyconf/access_loggers.go:111]: remove this

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/analytics/AnalyticsFilter.java [hunk 2]
```diff
@@ -155,6 +155,11 @@ public void handleSuccessRequest(RequestContext requestContext) {
                     endUserName == null ? APIConstants.END_USER_UNKNOWN : endUserName);
             requestContext.addMetadataToMap(MetadataConstants.API_CONTEXT_KEY,
                     requestContext.getMatchedAPI().getBasePath());
+
+            // Adding Gateway URL
+            String gatewayUrl = requestContext.getHeaders().get("x-original-gw-url");
```
Comment (VirajSalaka) [inline enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/analytics/AnalyticsFilter.java:160]: Intrude a constant in MetadataConstants

File: adapter/internal/oasparser/envoyconf/access_loggers.go (hunk 1 above)
Comment (Lasith650) [reply adapter/internal/oasparser/envoyconf/access_loggers.go]: Fixed with: be7bc20383c53e5ecb044bd929e2d385420e82a0

**Review by VirajSalaka — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : dev-deployment-v2 : 20230623.1

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : dev-deployment-v2 : 20230623.1

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : dev-deployment-v2 : 20230623.1

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : stage-deployment-v2 : 20230623.1

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20230623.1

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : stage-deployment-v2 : 20230623.1

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : stage-deployment-v2 : 20230627.1

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20230627.1

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : stage-deployment-v2 : 20230627.1

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20230628.1

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : prod-deployment-v2 : 20230628.1

**Pull request merged**